earthfile2llb/listener.go

```diff
@@ -962,6 +968,16 @@ func (ssf *StringSliceFlag) Set(arg string) error {
 	return nil
 }
 
+var shellNameStringRegexp = regexp.MustCompile("^[a-zA-Z_]+[a-zA-Z0-9_]*")
```

Comment:
```suggestion
var shellNameStringRegexp = regexp.MustCompile("^[a-zA-Z_]+[a-zA-Z0-9_]*$")
```

Make sure we match on the entire string only.